perses/tests/test_topology_proposal.py

```diff
@@ -72,6 +74,57 @@ def test_small_molecule_proposals():
         assert smiles == proposal.new_chemical_state_key
         proposal = new_proposal
 
+
+def test_small_molecule_constraint_repair_mapping():
+    """
+    Tests that a constrained edge is demapped after transformed to an unconstrained one.
+
+    This is done by checking that the new_to_old atom map doesn't have the atom that otherwise would be included.
+
+    For the ligands in the test this atom is the ONLY Fluorine atom involved in the transformation.
+    """
+    # TODO: we could try testing a pyridine to a fluoropyridine transformation with using the given geometries
+    from perses.utils.openeye import createOEMolFromSDF
+    # Get sdfs paths
+    sdf_path = resource_filename(
+        "perses",
+        os.path.join("data", "constrained-to-unconstrained", "ligands_constraint_test.sdf"),
+    )
+    # Using troublesome molecules in SDF data
+    ligand_a = createOEMolFromSDF(
+        sdf_path,
+        index=0)
+    ligand_b = createOEMolFromSDF(
+        sdf_path,
+        index=1)
+    # Set title to 'MOL'
+    ligand_a.SetTitle('MOL')
+    ligand_b.SetTitle('MOL')
+    # Create system
+    molecules = [ligand_a, ligand_b]
+    off_molecules = [Molecule.from_openeye(molecule) for molecule in molecules]
+
+    system_generator = SystemGenerator(forcefields=forcefield_files, barostat=barostat,
+                                       forcefield_kwargs=forcefield_kwargs,
+                                       nonperiodic_forcefield_kwargs=nonperiodic_forcefield_kwargs,
+                                       small_molecule_forcefield=small_molecule_forcefield, molecules=off_molecules,
+                                       cache=None)
+    # Propose topology
+    proposal_engine = SmallMoleculeSetProposalEngine(molecules, system_generator)
+    proposal_engine.use_given_geometries = True  # Force to use the given geometries in the propose
+    proposal_engine.given_geometries_tolerance = 0.4 * unit.angstrom
+    initial_system, initial_positions, initial_topology = OEMol_to_omm_ff(molecules[0], system_generator)
+    # NOTE: We need to run the propose method for constraints repairs to happen
+    proposal = proposal_engine.propose(initial_system, initial_topology)
+    # Find index for the fluorine atom is in the new molecule
+    for atom_idx, oe_atom in enumerate(list(ligand_b.GetAtoms())):
```

Comment:
This could be
```python
atoms = { atom.name : atom_index for atom_index, atom in enumerate(off_molecules[1].atoms) }
fluorine_index = atoms['F1']
```

Reply:
Using openff `Molecule`s can make it cleaner here, it would be a bit more intricate than that, since the aton names are not filled automatically when creating openff `Molecule` objects. We instead could use the `atom.element.symbol` or `atom.element.name` attributes, for `openff-toolkit` < 0.11. For >=0.11 we could use `atom.symbol` directly.

Reply:
So far `OEMol` is still needed, as per https://github.com/choderalab/perses/pull/1125#discussion_r1017131327

Reply:
> Using openff Molecules can make it cleaner here, it would be a bit more intricate than that, since the atom names are not filled automatically when creating openff Molecule objects

I totally forgot that SDF files don't contain atom names! Good point here.